### PR TITLE
rmdLibrary checks for installed pkg

### PR DESCRIPTION
**Pull request by MiGraber**
If a package is not installed then it will be installed and if not found throw a nice error

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -39,7 +39,11 @@ makeS3Obj2 = function(classes, object, ...){
 
 # Wrapper for loading not already loaded librarys
 rmdLibrary = function(needed.pkgs, file, force = FALSE){
-  catf("require(%s)\n", needed.pkgs, file = file)
+  catf('if (!require("%s",character.only = TRUE))
+    {
+    install.packages("%s",dep=TRUE)
+    if(!require("%s",character.only = TRUE)) stop("Package not found")\n
+    }', needed.pkgs, needed.pkgs, needed.pkgs, file = file)
```
Comment (tuanle618) [inline R/utils.R]: why do you add needed.pkgs three times? I assume this to be a vector

**Comment by tuanle618** [comment]
when executing:
`my.task = makeCorrTask(id = "test", data = cars)`
`my.corr = makeCorr(my.task)`
`report1 = makeCorrReport(my.corr, type = "CorrPlot")`
`finishReport(report1)`

I get following .rmd file-code:
`if (!require("stats",character.only = TRUE))`
  `  {`
   ` install.packages("stats",dep=TRUE)`
  `  if(!require("stats",character.only = TRUE)) stop("Package not found")`

    }`if (!require("GGally",character.only = TRUE))`
 `   {`
   ` install.packages("GGally",dep=TRUE)`
  `  if(!require("GGally",character.only = TRUE)) stop("Package not found")`

   ` }`

which cant be knitted. Seems to be some error with `}` . 

> Quitting from lines 2-14 (CorrReport1.rmd) 
Fehler in parse(text = x, srcfile = src) : <text>:6:6: Unerwartete(s) 'if'
5: 
6:     }if
        ^

**Comment by tuanle618** [comment]
OK, I see why you added it three times.. Somehow we need to pass over more args, otherwise catf returns an error.

This code seems to work:
`rmdLibrary = function(needed.pkgs, file, force = FALSE){`
`  catf("if(!require('%s', character.only = TRUE)) {`
`    install.packages('%s', dep = TRUE)`
`    stop(paste('Package', '%s' , 'not found'))`
`}\n", needed.pkgs, needed.pkgs, needed.pkgs, file = file)`
`}`

But I don't know why this is like that with the three times adding in the function. I'd suggest using a for loop and then writeLines(...) rather than catf in this case

**Comment by MiGraber** [comment]
You are right. I tested it only with a single string.
```
data("Arthritis", package = "vcd")
corr.task = makeCorrTask(data = Arthritis,id = "Corr.Task")
corr = makeCorr(corr.task)
corr.Report = makeCorrReport(corr, type = "CorrPlot")
finishReport(corr.Report, save.mode = FALSE)
```
this should work now

**Comment by tuanle618** [comment]
Ok. Please merge with master

**Pull request merged**